### ISSUE TITLE
Update repository deletion tests to observe new TASK returned

### ISSUE THREAD

**Issue by omaciel** [SatelliteQE/robottelo#1960]
Once https://github.com/Katello/katello/pull/5067 is merged we will have to update our existing API tests for deleting repositories to poll the returned task until completion/timeout.

**Comment by Ichimonji10** [comment]
The `EntityDeleteMixin.delete` method already checks the status code of the server's response and polls the returned ForemanTask if an HTTP ACCEPTED status code is returned. See [here](https://github.com/SatelliteQE/robottelo/blob/master/robottelo/orm.py#L262-L263). When that issue is closed, though, it definitely is a good idea to `git grep 1166365` and make sure that all relevant code is sane.

**Comment by omaciel** [comment]
This makes me **really** happy!!!

![leonard-dicaprio-dancing](https://cloud.githubusercontent.com/assets/53362/6465105/d8b3008a-c18c-11e4-8c8d-d63e7c04af8a.gif)

**Comment by Ichimonji10** [comment]
:D

**Comment by Ichimonji10** [comment]
Method `EntityDeleteMixin.delete` polls for completed tasks already. From `nailgun.entity_mixins`:

```python
        response = self.delete_raw()
        response.raise_for_status()
        if (synchronous is True and
                response.status_code == http_client.ACCEPTED):
            return _poll_task(response.json()['id'], self._server_config)
```
